tests/phpunit/CRM/MembershipExtras/Hook/PostProcess/UpdateSubscriptionTest.php

```diff
@@ -0,0 +1,476 @@
+use CRM_MembershipExtras_Test_Fabricator_PriceField as PriceFieldFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PriceFieldValue as PriceFieldValueFabricator;
+
+/**
+ * Class CRM_MembershipExtras_Hook_PostProcess_UpdateSubscriptionTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Hook_PostProcess_UpdateSubscriptionTest extends BaseHeadlessTest {
+
+  /**
+   * Contact for the payment plan.
+   *
+   * @var array
+   */
+  private $contact;
+
+  /**
+   * Membership type to use on tests.
+   *
+   * @var \CRM_Member_BAO_MembershipType
+   */
+  private $membershipType;
+
+  /**
+   * Parameters for the recurring contribution.
+   *
+   * @var array
+   */
+  private $recurringContributionParams = [];
+
+  /**
+   * Parameters for line items.
+   *
+   * @var array
+   */
+  private $lineItemsParams = [];
+
+  /**
+   * Parameters for first installment.
+   *
+   * @var array
+   */
+  private $contributionParams = [];
+
+  /**
+   * Data for the "Member Dues" financial type.
+   *
+   * @var array
+   */
+  private $memberDuesFinancialType = [];
+
+  /**
+   * Payment instrument ID for EFT.
+   *
+   * @var int
+   */
+  private $eftPaymentInstrumentID = 0;
+
+  /**
+   * ID of 'Pending' Contribution Status.
+   *
+   * @var int
+   */
+  private $contributionPendingStatusValue = 0;
+
+  /**
+   * Data for the default membership price set.
+   *
+   * @var array
+   */
+  private $defaultMembershipsPriceSet = [];
+
+  /**
+   * The form used to update recurring contributions.
+   *
+   * @var CRM_Contribute_Form_UpdateSubscription
+   */
+  private $updateSubscriptionForm;
+
+  /**
+   * Sets up common data required to run tests.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function setUp() {
+    $this->loadParameters();
+    $this->fabricateRequiredEntities();
+    $this->setUpDefaultPaymentPlanParameters();
+    $this->setUpUpdateSubscriptionForm();
+  }
+
+  /**
+   * Loads parameters required for the tests.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function loadParameters() {
+    $this->contributionPendingStatusValue = $this->getPendingContributionStatusValue();
+    $this->memberDuesFinancialType = $this->getMembershipDuesFinancialType();
+    $this->eftPaymentInstrumentID = $this->getEFTPaymentInstrumentID();
+    $this->defaultMembershipsPriceSet = $this->getDefaultPriceSet();
+  }
+
+  /**
+   * Obtains value for the 'Pending' contribution status option value.
+   *
+   * @return array
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function getPendingContributionStatusValue() {
+    return civicrm_api3('OptionValue', 'getvalue', [
+      'return' => 'value',
+      'option_group_id' => 'contribution_status',
+      'name' => 'Pending',
+    ]);
+  }
+
+  /**
+   * Obtains default price set for memberships.
+   *
+   * @return array
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function getDefaultPriceSet() {
```

Comment:
@omarabuhussein  @MiyaNoctem  I think this should be in the fabricator or at leat helper class.

Reply:
there is nothing getting created  here so no point in putting it in a fabricator, I am fine to keep  it here for the time being and if it happen we need it somewhere else then we can think were to put it.

Reply:
It could be part of the PriceField fabricator, but this one is kind of important for membership extras, as it is required to do autorenewal of plans with new memberships. Since the membership doesn't exist yet, the only way to determine the membership type is by looking at the price field value of the line item, which in turn should be associated to the default price set. So I thought it better to leave it explicit in the test case, and not hidden within the price set fabricator.